packages/twenty-front/src/modules/favorites/hooks/useReorderFavorite.ts

```diff
@@ -11,28 +13,106 @@ export const useReorderFavorite = () => {
   const { updateOneRecord: updateOneFavorite } = useUpdateOneRecord({
     objectNameSingular: CoreObjectNameSingular.Favorite,
   });
+  const setActiveFavoriteFolderId = useSetRecoilState(
+    activeFavoriteFolderIdState,
+  );
 
   const handleReorderFavorite: OnDragEndResponder = (result) => {
-    if (!result.destination) return;
+    const { destination, source, draggableId } = result;
 
-    const draggedFavoriteId = result.draggableId;
-    const draggedFavorite = favorites.find((f) => f.id === draggedFavoriteId);
+    if (!destination) return;
 
+    if (
+      destination.droppableId === source.droppableId &&
+      destination.index === source.index
+    ) {
+      return;
+    }
+
+    const draggedFavorite = favorites.find((f) => f.id === draggableId);
     if (!draggedFavorite) return;
 
-    const inSameFolderFavorites = favoritesSorted.filter(
-      (fav) => fav.favoriteFolderId === draggedFavorite.favoriteFolderId,
+    if (destination.droppableId.startsWith('folder-header-')) {
```

Comment:
please add tests on this hook
the naming of the hook should reflect that is is used for Drag and Drop as we are maniuplating droppable (and we have strong hypothesis on how the droppableId should be formed)

Reply:
can we also make this function throw in the case the droppableId is malformed